### PR TITLE
[JBIDE-22314] Allow defining an image registry URL for a connection

### PR DESCRIPTION
**Pull request by jcantrill**
cc @fbricon

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/URLValidator.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	public URLValidator(String urlType, boolean allowEmpty) {
+		invalidURLMessage = NLS.bind("Please provide a valid {0} URL", urlType);
+		this.allowEmpty  = allowEmpty;
+	}
+	
+	
+
+	@Override
+	public IStatus validate(Object in) {
+		String value = (String) in;
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/URLValidator.java]: There is a org.apache.commons.validator.routines.UrlValidator provided by commons-validator

Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/validator/URLValidator.java]: Updated

**Comment by fbricon** [comment]
I'll wait for the next sprint to merge this one, since there's nothing yet consuming the new registry url

**Comment by fbricon** [comment]
So whatever I type in the text field, I keep getting the "Please provide a valid image registry url" error message and the wizard's finish button stays disabled

**Comment by fbricon** [comment]
Ok so I was missing the https scheme (was using http).
- are we 100% sure no docker registry can be exposed through http? (think internal company docker registry)
- if yes, then the error message should be clearer, state only https connections are supported

**Comment by fbricon** [comment]
Other problem, if you have 2 connections in the wizard, or try to create a new connection, whatever you type in the advanced section is not reset when switching the connection.

**Comment by jcantrill** [comment]
@fbricon  updated PR to address your comment and issue

**Comment by fbricon** [comment]
+1

**Pull request merged**